code/datums/components/bump_attack.dm

```diff
@@ -2,34 +2,38 @@
 	var/active = TRUE
 	var/bump_action_path
 	var/datum/action/bump_attack_toggle/toggle_action
+	var/mob/living/bumper
```

Comment:
You're not supposed to hold a casted reference to a component parent, because this removes some genericty.
That being said, are bumped attacked supposed to be used by something not a mob/living? I don't think it will ever be the case, so why no. Not sure Tivi will like this though

Reply:
I think the core signals and checks used for the component require it to be on a living being, so if it was used for object, it would need to be rewritten anyways. 

I added this because for some reason "var/mob/living/bumper = parent" on each proc caused runtimes when using the component on humans.

Reply:
removed var